partition_manager.py: Fix a bug where 'align' spec was deleted. (#12031)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * The size of the span partitions was changed to include the alignment paritions (``EMPTY_x``) appearing between other partitions, but not alignment partitions at the beginning or end of the span partition.
    The size of the span partitions now reflects the memory space taken from the start of the first of its elements to the end of the last, not just the sum of the sizes of the included partitions.

>   * Fixed a bug where 'align' spec was deleted.

Reviewer: ```suggestion
  * Fixed a bug where the ``align`` spec was deleted.
```

>     This would happen in cases where two 'placement' specs were identical.

Reviewer: ```suggestion
    This would happen in cases where two ``placement`` specs were identical.
```

>     When disambiguating one of them, the 'align' spec was not preserved.

Reviewer: ```suggestion
    When disambiguating one of them, the ``align`` spec was not preserved.
```
